app/src/main/java/com/example/searchmovies/logic/MovieViewModel.kt

```diff
@@ -27,12 +28,35 @@ class MovieViewModel : ViewModel() {
 
     private fun searchMovie(searchText: String) {
         disposable.add(
-        searchMoviesUseCase.perform(searchText)
-            .subscribeOn(Schedulers.io())
-            .map { movies -> movies.map { MovieViewEntity("${it.title} - ${it.releaseDate}") } }
-            .observeOn(AndroidSchedulers.mainThread())
-            .subscribe { movieViewEntities -> searchedMoviesLiveData.value = MoviesContentViewEntity(movieViewEntities) }
+            searchMoviesUseCase.perform(searchText)
+                .subscribeOn(Schedulers.io())
+                .map { movies ->
+                    val theBestMovie = findTheBestMovie(movies)
+                    MoviesViewEntity(
+                        movies.joinToString { "${it.title} - ${it.releaseDate} " },
+                        TheBestMovieViewEntity(
+                            theBestMovie.popularity,
```

Comment:
O tym jeszcze nie mówiliśmy, ale Kotlin posiada "scope functions" - jak chcesz, to możesz pogooglować. W tym przypadku, żeby nie pisać wielokrotnie `theBestMovie.` można użyć 
`with(theBestMovie) {
TheBestMovieViewEntity(popularity, voteAverage, itp.)
}
`